in_toto/model_test.go

```diff
@@ -768,10 +769,9 @@ func TestValidatePubKey(t *testing.T) {
 		Scheme: "rsassa-pss-sha256",
 	}
 
-	err = validateRSAPubKey(testKey)
-	if err.Error() != "in key '776a00e29f3559e0141b3b096f696abc6cfb0c657ab40"+
-		"f441132b345b08453f5': private key found" {
-		t.Error("validateRSAPubKey error - private key not detected")
+	err = validateKey(testKey)
+	if !errors.Is(err, ErrEmptyKeyField) {
+		t.Error("validateKey error - private key not detected")
```

Comment:
I think this one also fails because of the missing `KeyIdHashAlgorithms` field, and not because of the "invalid" private key, right?

Reply:
There seems to be more of that below...

Reply:
This is also correct.. These tests were artifacts from older tests, looks like I just adopted them, without checking them more carefully :S 

What do we do with these now?

Reply:
Ideally we test what was tested before. For `ErrEmptyKeyField` errors this would mean to also take a look at the error message, in order to see which field was empty

Reply:
Makes sense. We could build error structs for this, like described here: https://blog.golang.org/go1.13-errors
or we wrap the key field into the error. What would you prefer? I am not sure if we could still use `errors.Is` with such error structs. I need to test this, but structs would be the more native way, I guess.

Reply:
I think we should try to wrap up this PR. Let's just check for error type `&&` string for now, and think about more sophisticated error handling in a follow up PR. :)

Reply:
I checked out the code:
```
	if key.KeyType == "" {
		return fmt.Errorf("%w: keytype", ErrEmptyKeyField)
	}
	if key.KeyVal.Public == "" {
		return fmt.Errorf("%w: keyval.public", ErrEmptyKeyField)
	}
	if key.Scheme == "" {
		return fmt.Errorf("%w: scheme", ErrEmptyKeyField)
	}
```
we actually do wrap the error here, already. Do you still want to check if the field got cached correctly? If so, we would need to do string compares again :S

Reply:
We can take another look at these in #68.